### PR TITLE
CI: Add Docker-based compatibility tests 

### PR DESCRIPTION
**Pull request by mkannwischer**
This PR adds Docker-based compatibility tests for various Ubuntu-20 and Ubuntu-22  docker images; so far, AArch64 only.

A number of changes had to be made, esp. to the test scripts; see commits.

Fix https://github.com/pq-code-package/mlkem-native/issues/560 

(Hoisted out from #562)

### CODE REVIEW

File: scripts/lib/mlkem_test.py [hunk 1]
```diff
@@ -355,7 +408,8 @@ def _func(opt: bool):
             exit(1)
 
     def _run_nistkat(self, opt: bool):
-        def expect_proc(scheme: SCHEME, actual: str) -> tuple[bool, str]:
+        def expect_proc(scheme: SCHEME, actual: str):
```
Comment (mkannwischer) [inline scripts/lib/mlkem_test.py:411]: In other places you removed the `:str` type information. 
Should that be removed in all places?

File: mk/config.mk [hunk 2]
```diff
@@ -13,11 +13,10 @@ CC  ?= gcc
...
 # NOTE: gcc-ar is a wrapper around ar that ensures proper integration with GCC plugins,
-# 		such as lto. Using gcc-ar is preferred when creating or linking static libraries
-# 		if the binary is compiled with -flto.
-# 		However, this doesn't apply to darwin as it is using clang instead, and there's no
-# 		gcc-ar wrapper as well.
-CC_AR ?= $(if $(findstring Darwin,$(shell uname -s)),ar,gcc-ar)
+# 	such as lto. Using gcc-ar is preferred when creating or linking static libraries
+# 	if the binary is compiled with -flto. However, it is not universally present, so
+#       only use it if available.
+CC_AR ?= $(if $(shell which gcc-ar),gcc-ar,ar)
```
Comment (mkannwischer) [inline mk/config.mk:19]: Doesn't this break if you are using Darwin with gcc-ar installed, but you compile with clang?
I'm not very sure about this.

Comment (hanno-becker) [reply mk/config.mk:19]: I tried to fix this by looking at `cc --version` to see if it's clang or not, and if it's gcc, check for gcc-ar.

File: scripts/lib/mlkem_test.py (hunk 1 above)
Comment (hanno-becker) [reply scripts/lib/mlkem_test.py:411]: Fixed (hopefully)

**Comment by mkannwischer** [comment]
Included in #562

**Pull request closed without merging**